src/Microsoft.DotNet.Interactive.HttpRequest.Tests/ParserTests.cs

```diff
@@ -0,0 +1,595 @@
+                  .MethodNode.Text.Should().Be(method);
+        }
+
+        [Fact]
+        public void http_version_is_parsed_correctly()
+        {
+            var result = HttpRequestParser.Parse("GET https://example.com HTTP/1.1");
+
+            result.SyntaxTree.RootNode
+                  .ChildNodes.Should().ContainSingle<HttpRequestNode>().Which
+                  .VersionNode.Text.Should().Be("HTTP/1.1");
+        }
+        [Fact]
+        public void diagnostic_object_is_reported_for_unrecognized_verb()
+        {
+            var result = HttpRequestParser.Parse("OOOOPS https://example.com");
+
+            result.GetDiagnostics()
+                .Should().ContainSingle().Which.Message.Should().Be("Unrecognized HTTP verb OOOOPS");
+        }
+
+        [Fact]
+        public void request_node_without_method_node_created_correctly()
+        {
+            var result = HttpRequestParser.Parse("https://example.com");
+
+            result.SyntaxTree.RootNode.ChildNodes.Should().ContainSingle<HttpRequestNode>().Which
+                .MethodNode.Should().BeNull();
+        }
+    }
+
+    public class Headers
+    {
+        [Fact]
+        public void headers_are_parsed_correctly()
+        {
+            var result = HttpRequestParser.Parse(
+                """
+                GET https://example.com HTTP/1.1
+                Accept: */*
+                Accept-Encoding : gzip, deflate, br
+                Accept-Language : en-US,en;q=0.9
+                ContentLength:7060
+                Cookie: expor=;HSD=Ak_1ZasdqwASDASD;SSID=SASASSDFsdfsdf213123;APISID=WRQWRQWRQWRcc123123;
+                """);
+
+            using var _ = new AssertionScope();
+
+            var headersNode = result.SyntaxTree.RootNode
+                  .ChildNodes.Should().ContainSingle<HttpRequestNode>().Which
+                  .ChildNodes.Should().ContainSingle<HttpHeadersNode>().Which;
+
+            var headerNodes = headersNode.HeaderNodes.ToArray();
+            headerNodes.Should().HaveCount(5);
+
+            headerNodes[0].NameNode.Text.Should().Be("Accept");
+            headerNodes[0].SeparatorNode.Text.Should().Be(":");
+            headerNodes[0].ValueNode.Text.Should().Be("*/*");
+
+            headerNodes[1].NameNode.Text.Should().Be("Accept-Encoding");
+            headerNodes[1].SeparatorNode.Text.Should().Be(":");
+            headerNodes[1].ValueNode.Text.Should().Be("gzip, deflate, br");
+
+            headerNodes[2].NameNode.Text.Should().Be("Accept-Language");
+            headerNodes[2].SeparatorNode.Text.Should().Be(":");
+            headerNodes[2].ValueNode.Text.Should().Be("en-US,en;q=0.9");
+
+            headerNodes[3].NameNode.Text.Should().Be("ContentLength");
+            headerNodes[3].SeparatorNode.Text.Should().Be(":");
+            headerNodes[3].ValueNode.Text.Should().Be("7060");
+
+            headerNodes[4].NameNode.Text.Should().Be("Cookie");
+            headerNodes[4].SeparatorNode.Text.Should().Be(":");
```

Comment:
Now that we have a `header_separator_is_present` test, we can remove the 4 assertions that validate the separator node's text from this test.